src/net_processing.cpp

```diff
@@ -2984,6 +2987,8 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         }
 
         std::list<CTransactionRef> lRemovedTxn;
+        PeerRef peer = GetPeerRef(pfrom.GetId());
```

Comment:
In commit "Move m_orphan_work_set to net_processing": I think you should put this call at the top of `ProcessMessage()` in this commit, rather than putting it here, and then moving it up in a later commit.

Alternatively, you could change the `ProcessMessage()` signature to take a `Peer&`. I don't think that's necessary, but you may prefer it to fetching the `PeerRef` in `ProcessMessages()` and then again in `ProcessMessage()`.